doc/scaling_spire.md

```diff
@@ -0,0 +1,43 @@
+# Scalability
+
+A SPIRE deployment is composed of a number one or more SPIRE Servers that share a replicated datastore, and one or more SPIRE Agents. With only one SPIRE server per SPIRE deployment, the memory and CPU usage tends to grow proportionally with the number of workloads in the deployment. To support very large numbers of workloads and agents within a given trust domain (tens of thousands or hundreds of thousands of nodes), the number of SPIRE Servers can be scaled horizontally. With multiple servers, the load is distributed. In addition to capacity, multiple SPIRE Servers can be configured for high availability.
+
+It is important to note when scaling the number servers horizontally, be it for high availability or load distribution purposes, that in any configuration of two or more SPIRE servers, all servers share the same datastore, while each individual server maintains an independent root certificate/key. 
+
+# Topologies
+
+There are three main SPIRE deployment topologies; single trust domain, nested SPIRE, and federated SPIRE. Multiple considerations should are to be factored in when choosing one topology design over another.
+
+A common motivation for a single overarching trust domain is to issue identifies from a single Certificate Authority, as it reduces the number of components to manage. However, when deploying a single SPIRE trust domain to span regions and clouds. 
+Managing a shared datastore across geographically dispersed locations or across cloud providers may represent a challenge. In such circumstances, a solution is to configure SPIRE servers can be configured in Nested topology.
+
+## Nested SPIRE
+
+Nested SPIRE allows SPIRE servers to be chained together, and for all servers to still issue identities in the same trust domain. It works by co-locating an agent with downstream servers. Each downstream server loads the CA credentials from the Workload API of the top or global SPIFFE implementation. The downstream node performs both node and workload attestation, which in turn allows automating the bring-up of new downstream servers. 
```

Comment:
This description isn't quite accurate. The downstream SPIRE server obtains credentials over the Workload API that it uses to _directly_ authenticate with the upstream SPIRE server to obtain an intermediate CA.

Also, I couldn't quite grok the last sentence. What is the purpose of talking about node attestation here? I'm not following something :grimacing:

Reply:
Thanks. I added the order of operations here.